frontend/public/components/storage-class-form.tsx

```diff
@@ -0,0 +1,927 @@
+          <p>
+            Specific fields for the selected provisioner. &nbsp;
+            <Link to={this.storageTypes[this.state.newStorageClass.type].documentationLink} target="_blank">
+              What should I enter here? <i className="fa fa-external-link" />
+            </Link>
+          </p>
+          <NameValueEditor
+            nameValuePairs={this.state.customParams}
+            nameString="Parameter"
+            valueString="Value"
+            addString="Add Parameter"
+            updateParentData={this.updateCustomParams} />
+        </FormGroup>
+      </React.Fragment>
+    );
+  };
+
+  close = () => {
+    this.props.stopK8sWatch(this.reduxId);
+    if (typeof this.props.onClose === 'function') {
+      this.props.onClose();
+    }
+    history.push('/k8s/cluster/storageclasses');
+  };
+
+  render() {
+    const newStorageClass = this.state.newStorageClass;
+    const reclaimPolicyKey = newStorageClass.reclaim === null ? this.reclaimPolicies.Delete : newStorageClass.reclaim;
+    const fieldErrors = this.state.fieldErrors;
+
+    return (
+      <div className="co-m-pane__body">
+        <Form className="wizard-form">
+          <h1 className="co-m-pane__heading">Create Storage Class</h1>
+
+          <FormGroup controlId={'basic-settings-name'} validationState={fieldErrors.nameValidationMsg ? 'error': null}>
+            <label className="control-label co-required" htmlFor="storage-class-name">Name</label>
+            <FormControl
+              type="text"
+              placeholder={newStorageClass.name}
+              id="storage-class-name"
+              // onChange={(event) => this.setNameHandler(event)}
+              onChange={(event) => this.setStorageHandler('name', event.target.value)}
+              value={_.get(newStorageClass, 'name', '')} />
+            <HelpBlock>{fieldErrors.nameValidationMsg ? fieldErrors.nameValidationMsg : null}</HelpBlock>
+          </FormGroup>
+
+          <FormGroup controlId={'basic-settings-description'}>
+            <label htmlFor="storage-class-description">Description</label>
+            <FormControl
+              type="text"
+              id="storage-class-description"
+              onChange={(event) => this.setStorageHandler('description', event.target.value)}
+              value={_.get(newStorageClass, 'description', '')} />
+          </FormGroup>
+
+          <FormGroup controlId={'basic-settings-reclaim-policy'}>
+            <label className="control-label co-required">Reclaim Policy</label>
+            <Dropdown
+              title="Choose Reclaim Policy"
+              items={this.reclaimPolicies}
+              dropDownClassName="dropdown--full-width"
+              selectedKey={reclaimPolicyKey}
+              onChange={(event) => this.setStorageHandler('reclaim', event)} />
+            <HelpBlock>
+              Determines what happens to persistent volumes when the associated persistent volume claim is deleted. Defaults to &lsquo;Delete&rsquo;
```

Comment:
Strange that this is marked as required, but it has a default value. (At least the label has the `co-required` class.)

Reply:
I did that based off the design, but it's removed now. @cshinn I'm assuming it was just an oversight, but let me know if it should be added back for some reason.

Reply:
Yeah, sorry. Definitely go with what's actually required by K8s over the data in the mockups